src/baseline.rs

```diff
@@ -7,22 +8,196 @@ use crate::dump::RustDocCommand;
+    // outside and remove this TODO.
+    //
+    // Implicit features from optional dependencies have to be added separately
+    // from regular features: https://github.com/obi1kenobi/cargo-semver-checks/issues/265
+    let mut implicit_features: BTreeSet<_> = crate_
+        .dependencies()
+        .iter()
+        .filter_map(|dep| dep.is_optional().then_some(dep.name()))
+        .map(|x| x.to_string())
+        .collect();
+    for feature_defn in crate_.features().values().flatten() {
+        // "If you specify the optional dependency with the dep: prefix anywhere
+        //  in the [features] table, that disables the implicit feature."
+        // https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies
+        if let Some(optional_dep) = feature_defn.strip_prefix("dep:") {
+            implicit_features.remove(optional_dep);
+        }
+    }
+    let regular_features: BTreeSet<_> = crate_.features().keys().cloned().collect();
+    let mut all_crate_features = implicit_features;
+    all_crate_features.extend(regular_features);
+    all_crate_features.into_iter().collect()
+}
+
+#[allow(unused_variables)]
+fn get_all_crate_features_from_manifest(path: &Path) -> Vec<String> {
+    unimplemented!()
+}
+
+#[allow(dead_code)]
+enum CrateSource<'a> {
+    Registry { crate_: &'a crates_index::Version },
+    ManifestPath { path: &'a Path, name: String },
+}
+
+/// To get the rustdoc of the baseline, we first create a placeholder project somewhere
+/// with the baseline as a dependency, and run `cargo rustdoc` on it.
+fn create_placeholder_rustdoc_manifest(crate_source: &CrateSource) -> cargo_toml::Manifest<()> {
+    use cargo_toml::*;
+
+    Manifest::<()> {
+        package: {
+            let mut package = Package::new("rustdoc", "0.0.0");
+            package.publish = Inheritable::Set(Publish::Flag(false));
+            Some(package)
+        },
+        workspace: Some(Workspace::<()>::default()),
+        lib: {
+            let product = Product {
+                path: Some("lib.rs".to_string()),
+                ..Product::default()
+            };
+            Some(product)
+        },
+        dependencies: {
+            let project_with_features: DependencyDetail = match crate_source {
+                CrateSource::Registry { crate_ } => DependencyDetail {
+                    // We need the *exact* version as a dependency, or else cargo will
+                    // give us the latest semver-compatible version which is not we want.
+                    // Fixes: https://github.com/obi1kenobi/cargo-semver-checks/issues/261
+                    version: Some(format!("={}", crate_.version())),
+                    features: get_all_crate_features_from_registry(crate_),
+                    ..DependencyDetail::default()
+                },
+                CrateSource::ManifestPath { path, name: _ } => DependencyDetail {
```

Comment:
```suggestion
                CrateSource::ManifestPath { path, .. } => DependencyDetail {
```
Perhaps a bit easier to read? Your call.